Compatibility Issue with React 19 Release Candidate

I ran into an issue when trying to install `@heroicons/react` with the React 19 release candidate (`19.0.0-rc-69d4b800-20241021`). The installation fails due to a dependency conflict because `@heroicons/react` currently supports React versions `>= 16`, which doesn’t appear to include compatibility with React 19.

**Error Details**

```bash
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: my-next15-app@0.1.0
npm ERR! Found: react@19.0.0-rc-69d4b800-20241021
npm ERR! node_modules/react
npm ERR!   react@"19.0.0-rc-69d4b800-20241021" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@">= 16" from @heroicons/react@2.1.5
npm ERR! node_modules/@heroicons/react
npm ERR!   @heroicons/react@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

**Steps to Reproduce**

1. Start a new Next.js project with React 19 (e.g., `npx create-next-app@latest`).
2. Attempt to install `@heroicons/react`:
   ```bash
   npm install @heroicons/react
   ```
3. The error appears due to the dependency conflict.

**Environment:**

- `@heroicons/react` version: 2.1.5
- React version: 19.0.0-rc-69d4b800-20241021
- Node version: 20.10.0
- npm version: 10.2.3

**Workarounds I Found:**
1. **Downgrade React** to version 18.

**Request:**
Any chance you could add support for the React 19 release candidates? It would make it easier for others exploring the latest React updates to use Heroicons!

Thanks so much!

i'm encountering the same issue 

**Error Details**

npm install @heroicons/react
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: andover@0.1.0
npm ERR! Found: react@19.0.0-rc-02c0e824-20241028
npm ERR! node_modules/react
npm ERR!   react@"19.0.0-rc-02c0e824-20241028" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@">= 16" from @heroicons/react@2.1.5
npm ERR! node_modules/@heroicons/react
npm ERR!   @heroicons/react@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

i'm encountering the same issue


**Error Details**

```
npm install @heroicons/react
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: andover@0.1.0
npm ERR! Found: react@19.0.0-rc-02c0e824-20241028
npm ERR! node_modules/react
npm ERR! react@"19.0.0-rc-02c0e824-20241028" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@">= 16" from @heroicons/react@2.1.5
npm ERR! node_modules/@heroicons/react
npm ERR! @heroicons/react@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

I've encountered the same issue. with Nextjs 15 and React 19. Installing with `--legacy-peer-deps` works.

```
npm i -S @headlessui/react@v2.2.0
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: @heroicons/react@2.1.5
npm error Found: react@19.0.0-rc-69d4b800-20241021
npm error node_modules/react
npm error   peer react@"^18.2.0 || 19.0.0-rc-69d4b800-20241021" from next@15.0.1
npm error   node_modules/next
npm error     peer next@"^14.0.0-0 || ^15.0.0-0" from next-auth@5.0.0-beta.25
npm error     node_modules/next-auth
npm error       next-auth@"^5.0.0-beta.25" from the root project
npm error     next@"15.0.1" from the root project
npm error   peer react@"^18.2.0 || ^19.0.0-0" from next-auth@5.0.0-beta.25
npm error   node_modules/next-auth
npm error     next-auth@"^5.0.0-beta.25" from the root project
npm error   5 more (react-dom, styled-jsx, use-debounce, the root project, @headlessui/react)
npm error
npm error Could not resolve dependency:
npm error peer react@">= 16" from @heroicons/react@2.1.5
npm error node_modules/@heroicons/react
npm error   @heroicons/react@"^2.1.5" from the root project
npm error
npm error Conflicting peer dependency: react@18.3.1
npm error node_modules/react
npm error   peer react@">= 16" from @heroicons/react@2.1.5
npm error   node_modules/@heroicons/react
npm error     @heroicons/react@"^2.1.5" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,9 +7,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Add React 19 support ([#1247](https://github.com/tailwindlabs/heroicons/pull/1247))
+
 ### Fixed
 
-- Removed unncessary clipping path from `solid/arrow-left-circle` ([#1211](https://github.com/tailwindlabs/heroicons/pull/1211))
+- Removed unnecessary clipping path from `solid/arrow-left-circle` ([#1211](https://github.com/tailwindlabs/heroicons/pull/1211))
 
 ## [2.1.5] - 2024-07-10
 

--- a/react/package.json
+++ b/react/package.json
@@ -144,6 +144,6 @@
     "access": "public"
   },
   "peerDependencies": {
-    "react": ">= 16"
+    "react": ">= 16 || ^19.0.0-rc"
   }
 }

